packages/core/src/hydration/interfaces.ts

```diff
@@ -165,6 +165,16 @@ export interface DehydratedView {
    * nodes detected in this view at serialization time.
    */
   disconnectedNodes?: Set<number>|null;
+
+  /**
+   * A mapping from a view to the first child to begin claiming nodes.
+   *
+   * This mapping is generated by an i18n block, and is the source of
+   * truth for the nodes inside of it. Once a view has been hydrated,
+   * the node is removed from the map. The remaining entries indicate
+   * dehydrated nodes that need to be cleaned up.
+   */
+  nodeMap?: Map<number, RNode>;
```

Comment:
Since this is i18n-only field, I'd propose including `i18n` into the name of the field (and corresponding functions that work with this field):
```suggestion
  i18nNodes?: Map<number, RNode>;
```